scripts/staging/fm-regression.dml

```diff
@@ -0,0 +1,181 @@
+#-------------------------------------------------------------
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+
+/**
+ * Factorization Machines for Regression.
+ */
+
+# Imports
+source("nn/optim/adam.dml") as adam
+source("nn/layers/cross_entropy_loss.dml") as cross_entropy_loss
+source("nn/layers/fm.dml") as fm
+source("nn/layers/l2_loss.dml") as l2_loss
+source("nn/layers/l2_reg.dml") as l2_reg
+source("nn/layers/log_loss.dml") as log_loss
+source("nn/layers/sigmoid.dml") as sigmoid
+
+n = 50 # num examples
+d = 500 # num features
+k = 2 # factorization dimensionality
+X = rand(rows=n, cols=d);
+y = rand(rows=n, cols=1);
+X_val = rand(rows=n, cols=d)
+y_val = rand(rows=n, cols=1)
```

Comment:
These should be moved to another "script" file that imports this file, sets up some real or dummy data, and calls the train function.  Similar to the LeNet or softmax examples in https://github.com/apache/systemml/tree/master/scripts/nn/examples.